<div class="container-fluid">
	<div class="row">
		<div class="col-md-12">
			<h3 >
				About it.
			</h3>
			<p>Bacterial and archaeal organisms respond to diverse stimuli across a subtle mechanism of regulation at transcriptional level, where DNA-binding transcriptional factors are involved. These proteins act by interacting with specific sites upstream the transcription start site, allowing or blocking the access to the RNA polymerase to the promoter. In general, when the TF binds overlapping the promoter, the system is repressed; however when the binding site is upstream the promoter, the system is activated. In addition, the system is coordinated with the sensing of endogenous or exogenous stimuli by these regulatory proteins; i.e. they have the ability to sense the diverse conditions in the cell to contend against the environmental changes. For instance, in the bacterium Escherichia coli K-12, around three-quarters of TFs could respond directly to extracellular signals through phosphorylation and small-molecule binding, such as allolactose or maltose, among others.</p> 
			<p>The regulatory system can be conceptualized as a circuit, where one TF can be regulating multiple target genes (TGs) and multiple genes can be regulated by one or diverse TFs, all of them assembled into a gene regulatory network (GRN). This GRN is a directed graph in which TFs are connected to their TGs by negative or positive interaction edges.</p>
			<p>To date, GRNs have been determined only for a few model organisms on three different phyla, such as the Proteobacteria, Escherichia coli K-12, Salmonella typhimurium, and Pseudomonas aeruginosa; the Firmicute, Bacillus subtilis 168, and Staphylococcus aureus N315; and the Actinobacteria, Mycobacterium tuberculosis, since their reconstruction depends largely on experimental data. Therefore, the inference or expansion of regulatory relationships between TFs and their TGs in organisms beyond the bacterial models will allow us to understand diverse biological processes, such as cell growth, response to environmental changes or cell division, among others.</p>
			<p>Hence, in order to determine the GRN in bacterial and archaeal genomes with no information of their regulatory networks, in our work we mapped orthologous interactions among the six bacterial models to identify interactions not previously described. Posteriorly, we used the approach of guilt by association to infer the GRNs for 5321 genomes, based on TF-TG orthology relationships of six bacterial species with well-known regulatory interactions, and Transcription Units assignments. The reconstructed networks were evaluated in terms of their topological properties, identifying TFs as hubs, modules and co-regulated genes. Therefore, our approach allows us to confer a degree of accuracy regarding the existence of each inferred interaction. Finally, through this web server we offer the posibility to study the regulatory networks above described.</p>
			</p>
		</div>
	</div>
</div>
